pkg/controller/argocd/secret.go

```diff
@@ -423,12 +440,22 @@ func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.Arg
 	}
 	for _, s := range clusterSecrets.Items {
 		// check if cluster secret with default server address exists
-		// do nothing if exists.
+		// update the list of namespaces if value differs.
 		if string(s.Data["server"]) == common.ArgoCDDefaultServer {
 			if clusterConfigInstance {
-				r.client.Delete(context.TODO(), &s)
+				if err := r.client.Delete(context.TODO(), &s); err != nil {
+					return err
+				}
```

Comment:
I was just wondering whether we want to only delete `namespaces` key from the secret here, instead of completely deleting the secret. User may have created secret to change other properties, such as cluster shard index. But let's make this another PR.

Reply:
Could there be any other configuration in the secret that could affect the functioning of a `ClusterConfig ArgoCD instance` ?